src/OSPSuite.Core/Domain/Populations/ParameterValuesCache.cs

```diff
@@ -0,0 +1,188 @@
+            return parameterPath;
+
+         var allSupportedPath = AllParameterPaths();
+         var index = Array.FindIndex(allSupportedPath, x => x.StartsWith(parameterPath));
+         //try to strip the unit and see if it's possible to find a key exactly equal to the parameterPath without the unit
+         if (index >= 0 && string.Equals(allSupportedPath[index].StripUnit(), parameterPath))
+            return allSupportedPath[index];
+
+         return string.Empty;
+      }
+
+      public virtual ParameterValues ParameterValuesFor(string parameterPath)
+      {
+         var possibleKey = internalKeyFor(parameterPath);
+         if (string.IsNullOrEmpty(possibleKey))
+         {
+            possibleKey = parameterPath;
+            Add(new ParameterValues(parameterPath));
+         }
+
+         return _parameterValuesCache[possibleKey];
+      }
+
+      public virtual ParameterValues ParameterValuesAt(int index) => _parameterValuesCache.ElementAt(index);
+
+      public double[] GetValues(string parameterPath) => ValuesFor(parameterPath).ToArray();
+
+      public virtual void Add(ParameterValues parameterValues) => _parameterValuesCache.Add(parameterValues);
+
+      private void addValues(IEnumerable<ParameterValue> parameterValues)
+      {
+         parameterValues.Each(pv => ParameterValuesFor(pv.ParameterPath).Add(pv));
+      }
+
+      public virtual bool Has(string parameterPath)
+      {
+         var possibleKey = internalKeyFor(parameterPath);
+         return !string.IsNullOrEmpty(possibleKey);
+      }
+
+      public virtual IReadOnlyList<double> ValuesFor(string parameterPath) => ParameterValuesFor(parameterPath).Values;
+
+      public virtual IReadOnlyList<double> PercentilesFor(string parameterPath) => ParameterValuesFor(parameterPath).Percentiles;
+
+      public virtual string[] AllParameterPaths() => _parameterValuesCache.Keys.ToArray();
+
+      public virtual ParameterValuesCache Clone()
+      {
+         var clone = new ParameterValuesCache();
+         AllParameterValues.Each(x => clone.Add(x.Clone()));
+         return clone;
+      }
+
+      public virtual void Remove(string parameterPath) => _parameterValuesCache.Remove(parameterPath);
+
+      public virtual void SetValues(string parameterPath, IReadOnlyList<RandomValue> newValues) => setValues(parameterPath, newValues, x => x.Add);
+
+      public virtual void SetValues(string parameterPath, IReadOnlyList<double> newValues) => setValues(parameterPath, newValues, x => x.Add);
+
+      private void setValues<T>(string parameterPath, IReadOnlyList<T> newValues, Func<ParameterValues, Action<IReadOnlyList<T>>> addValuesFunc)
+      {
+         validateCount(parameterPath, newValues);
+         var parameterValues = ParameterValuesFor(parameterPath);
+         parameterValues.Clear();
+         addValuesFunc(parameterValues)(newValues);
+      }
+
+      private void validateCount<T>(string parameterPath, IReadOnlyList<T> values)
+      {
```

Comment:
This is to check that we do not create columns that do not have all the values over all individuals( required when adding stuff manually from R)